src/plugins/safeSnap/utils/umaModule.ts

```diff
@@ -68,17 +73,99 @@ export const getModuleDetails = async (
     [moduleAddress, 'liveness']
   ]);
   // NOTE: Need to get questionId, finalizedAt, isApproved
+  let needsApproval = false;
+  const minimumBond = moduleDetails[3][0];
+  const optimisticOracle = moduleDetails[1][0];
+
   const bondDetails = await getBondDetails(provider, moduleAddress);
 
-  const minimumBond = moduleDetails[3][0];
-  let needsApproval = false;
   if (
     Number(minimumBond) > 0 &&
     Number(minimumBond) > Number(bondDetails.currentUserBondAllowance)
   ) {
     needsApproval = true;
   }
 
+  // Create ancillary data for proposal hash
+  let ancillaryData = '';
+  if (transactions !== undefined) {
+    const transactionsHash = keccak256(
+      defaultAbiCoder.encode(
+        ['(address to, uint8 operation, uint256 value, bytes data)[]'],
+        [transactions]
+      )
+    );
+
+    ancillaryData = pack(
+      ['string', 'bytes', 'bytes'],
+      [
+        '',
+        pack(['string', 'string'], ['proposalHash', ':']),
+        toUtf8Bytes(transactionsHash.replace('0x', ''))
+      ]
+    );
+  }
+
+  // Search for requests with matching ancillary data
+  const oracleContract = new Contract(
+    optimisticOracle,
+    UMA_ORACLE_ABI,
+    provider
+  );
+  const events = await oracleContract.queryFilter('ProposePrice', -10000);
+  const proposeEvent = events.filter(
```

Comment:
You could also filter by the proposal timestamp since we store that in the module, mapped to the proposal hash.

Reply:
I updated to include the proposal timestamp. Let me know if there is a better implementation.